diff --git a/ggml/src/ggml-backend.cpp b/ggml/src/ggml-backend.cpp
index eeaf35c..ea9728c 100644
--- a/ggml/src/ggml-backend.cpp
+++ b/ggml/src/ggml-backend.cpp
@@ -1631,7 +1631,7 @@ ggml_backend_sched_t ggml_backend_sched_new(
     sched->prev_leaf_backend_ids = (int *) calloc(nodes_size, sizeof(sched->prev_leaf_backend_ids[0]));
 
     sched->context_buffer_size = ggml_sched_max_splits*GGML_SCHED_MAX_SPLIT_INPUTS*2*sizeof(struct ggml_tensor) + ggml_graph_overhead_custom(graph_size, false);
-    sched->context_buffer = (char *) malloc(sched->context_buffer_size);
+    sched->context_buffer = (char *) ggml_aligned_malloc(sched->context_buffer_size);
 
     const int initial_splits_capacity = 16;
     sched->splits = (ggml_backend_sched_split *) calloc(initial_splits_capacity, sizeof(sched->splits[0]));
@@ -1676,7 +1676,7 @@ void ggml_backend_sched_free(ggml_backend_sched_t sched) {
     free(sched->leaf_backend_ids);
     free(sched->prev_node_backend_ids);
     free(sched->prev_leaf_backend_ids);
-    free(sched->context_buffer);
+    ggml_aligned_free(sched->context_buffer, sched->context_buffer_size);
     free(sched->graph.nodes);
     free(sched->graph.leafs);
     free(sched);
